types/readable-stream/index.d.ts

```diff
@@ -103,7 +110,7 @@ declare namespace _Readable {
     class PassThrough extends Transform implements stream.PassThrough {
         constructor(options?: TransformOptions);
 
-        _transform<T>(chunk: T, encoding: BufferEncoding | null | undefined, callback: (error: any, data: T) => void): void;
+        _transform<T>(chunk: T, encoding: BufferEncoding | string | null | undefined, callback: (error?: Error, data?: T) => void): void;
```

Comment:
Why `| string` since BufferEncoding covers valid node.js encoding type/strings?

Reply:
When I went to test these types on a sample ionic project, it would not compile:

```
node_modules/@types/readable-stream/index.d.ts:19:15 - error TS2417: Class static side 'typeof _Readable' incorrectly extends base class static side 'typeof Readable'.
  The types of 'Stream.Readable.Writable' are incompatible between these types.
    Type 'typeof _Readable.Writable' is not assignable to type 'typeof import("stream").Writable'.
      Types of parameters 'options' and 'opts' are incompatible.
        Type 'import("stream").WritableOptions' is not assignable to type '_Readable.WritableOptions'.
          Type 'WritableOptions' is not assignable to type 'WritableStateOptions'.
            Types of property 'defaultEncoding' are incompatible.
              Type 'string' is not assignable to type 'BufferEncoding'.

19 declare class _Readable extends stream.Readable {
```

 I then looked at the node types and even though BufferedEncoding is defined in global, the v10 types use string. 

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v10/stream.d.ts#L102

I considered making it string, but I figured it would be best to give newer clients the more specific type information while still maintaining backwards type compatibility.

Ionic is using this in the packages.json:
    "@types/readable-stream": "^2.3.7",
    "@types/node": "^13.13.12",